bedrock/newsletter/urls.py

```diff
@@ -43,7 +39,7 @@
         ),
         name="newsletter.knowledge-is-power",
     ),
-    re_path("^newsletter/knowledge-is-power/confirm/(?P<token>[^/]*)/?$", views.kip_confirm, name="newsletter.knowledge-is-power.confirm"),
+    path("newsletter/knowledge-is-power/confirm/<uuid:token>/", views.kip_confirm, name="newsletter.knowledge-is-power.confirm"),
```

Comment:
**issue:** do we need a `no-token` version of this page? When I load the page and hit refresh, I get a 404.

Same issue for `/country/` and `/confirm/`.

Reply:
Gotcha. Those had optional tokens that I missed, which is one reason why converting these to the Django path converter is nice. In the latest commits on this branch I've added the no-token variants.

The `confirm` view does require a token, however, since it uses it in the view, just just in the JS.